### PR TITLE
Osd 6923 make hibernation aware

### PR DESCRIPTION
**Pull request by sedroche**
Jira: https://issues.redhat.com/browse/OSD-6923 
Docs for hibernation: https://github.com/openshift/hive/blob/master/docs/hibernating-clusters.md

- This switches the pd service enabled/disable on powerState state.
- When hibernating any active alerts are cleared.

### CODE REVIEW

File: pkg/pagerduty/mock/mock_service.go [hunk 1]
```diff
@@ -65,11 +65,9 @@ func (mr *MockClientMockRecorder) GetIntegrationKey(data interface{}) *gomock.Ca
 }
 
 // CreateService mocks base method
-func (m *MockClient) CreateService(data *pagerduty.Data) (string, error) {
+func (m *MockClient) CreateService(data *pagerduty0.Data) (string, error) {
 	m.ctrl.T.Helper()
 	ret := m.ctrl.Call(m, "CreateService", data)
-	data.ServiceID = "XYZ123"
```
Comment (sedroche) [inline pkg/pagerduty/mock/mock_service.go:71]: I'm not sure how this was generated previously but I ran `mockgen -source=service.go -destination=./mock/mock_service.go` which sortof matches the current mockfile. This change seems to be added directly to the generated files though so I have moved it to

Worth getting this generated by boilerplate but im not 100% up on that process

File: pkg/controller/pagerdutyintegration/pagerdutyintegration_controller_test.go [hunk 2]
```diff
@@ -243,38 +248,52 @@ func TestReconcilePagerDutyIntegration(t *testing.T) {
...
 			localObjects: []runtime.Object{
-				testClusterDeployment(true, true, true, false),
+				testClusterDeployment(true, true, true, false, false),
 				testPDISecret(),
 				testPagerDutyIntegration(),
 			},
 			expectPDSetup: true,
 			setupPDMock: func(r *mockpd.MockClientMockRecorder) {
-				r.CreateService(gomock.Any()).Return(testIntegrationID, nil).Times(1)
+				r.CreateService(gomock.Any()).Return(testIntegrationID, nil).Times(1).DoAndReturn(
```
Comment (sedroche) [inline pkg/controller/pagerdutyintegration/pagerdutyintegration_controller_test.go:278]: See comment https://github.com/openshift/pagerduty-operator/pull/157/files#r690849779 for these changes

**Comment by wanghaoran1988** [comment]
This looks good to me generialy, @NautiluX Could you help take a look when you have time?

**Comment by jewzaam** [comment]
@sedroche @wanghaoran1988 anybody can give a lgtm and it'll merge via bot now that it is approved.

**Pull request merged**